src/lava/utils/plots.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def raster_plot(spks, stride=6, fig=None, color='b', alpha=1):
+    """Generate raster plot of spiking activity.
+
+    Parameters
+    ----------
+    spks : np.ndarray shape (num_neurons, num_timesteps)
+        Spiking activity of neurons, a spike is indicated by a one
+    stride : int
+        Stride for plotting neurons
+    """
+    num_neurons = spks.shape[0]
+    num_time_steps = spks.shape[1]
```

Comment:
Please add input validation here, where users receive meaningful error messages when `spikes` is specified in the wrong format.

Similarly, please add input validation for the remaining parameters.

I suggest writing unit tests for these cases that check whether the exceptions are raised in exactly the right circumstances.

Reply:
I added some validation. But there are two kinds that I left out for now:
- Validation of the parameters directly related to matplotlib, since we are merely forwarding them. In this case we might want to rely on the matplotlib-internal validation. Let me know what you think.
- Type checking validation: I saw that there are some cases in this project where this is done and type errors are raised. Is there a convention?

Reply:
I agree with leaving out both types of validation. Type checking is sometimes considered valuable by some but I tend to guide users with type annotations rather than add a lot of validation. That way, they can at least get some warning highlights in the IDE.